### ISSUE TITLE
Impossible to include non-breaking-space (&nbsp;) in HTML 

### ISSUE THREAD

**Issue by imeckler** [evancz/elm-html#54]
The title should be self explanatory. One cannot use `Html.text` since it escapes.

**Comment by ajevans85** [comment]
Yes, this is a big pain, any html escape codes will be escaped. It would be useful to have a Html.raw for such cases. I can't seem to see an easy way to do this but it's a bit limitation!

**Comment by ajevans85** [comment]
Ok, I have a hack, it works but it is a hack...

Looking elm-html relies on elm-virtualdom which is built on top of the native js https://github.com/Matt-Esch/virtual-dom.

When you use `Html.text` underneath it uses the browsers `document.createTextNode` function https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode . This is how we get our escaping for free.

We can't create a `node` with arbitrary html because it uses the browsers `document.createElement` and a `node` / `element` takes a list of elements, either a node or text.

But, we have a hack. https://github.com/Matt-Esch/virtual-dom/blob/88534bbd34c5befbd1689fc71e2373c9145a9807/docs/vtext.md .

We can create a node and then set the `innerHtml` property exposed by `virtual-dom`. The docs *strongly recommend against* doing this but its working for my simple string case. I guess for complex elements it may/will break.

With the below function: 

```
import Html(..) 
import Json.Encode as Json
import List exposing ((::))
import VirtualDom exposing (property, Property)

h3' : List Attribute -> String -> Html
h3' props html =
  let innerHtml = property "innerHTML" (Json.string html)
  in
      node "h3" (innerHtml :: props) []
```
We can now use raw html:
```
h3' [ class "my_class", id "my_id"] "Hello <span class=\"highlight\">world</span> &hellip;"
```

I've not tried but you might also be able to use a css trick, i'm not sure if the content will be escaped or not. Create an empty span:
```
<span class="copy"></span>
```

Create a css rule:
```
span.copy:after{
  content: "&amp";
}​
```

**Comment by rtfeldman** [comment]
How about using [the unicode non-breaking space character](http://www.fileformat.info/info/unicode/char/00a0/index.htm)?

(The Elm string `"\u00A0"` should do the trick.)

**Comment by wyattbenno777** [comment]
I tried that in html text "\u00A0" and got.
(line 1, column 3): unexpected "u" expecting space, "&" or escape code

):

**Comment by rtfeldman** [comment]
Yeah I could have sworn this used to work...maybe it stopped working in a later release?

At any rate, now I just copy/paste in an actual [unicode nonbreaking space](http://www.fileformat.info/info/unicode/char/00a0/index.htm) character. 😉

**Comment by lydell** [comment]
`"\x00a0"` and `"\x00A0"` work for me.

**Comment by rtfeldman** [comment]
Ahh, that's it! Nice!

**Comment by jasonmahr** [comment]
Hey guys, I made a [package](https://github.com/jasonmahr/html-escape-sequences/tree/1.0.0) that solves this. If you use text' "&nbsp;" it'll render the nonbreaking space instead of the escape sequence. Hope this helps!

**Comment by jasonmahr** [comment]
Hey guys, I made a [package](https://github.com/jasonmahr/html-escape-sequences/tree/1.0.0) that solves this. If you use `text' "&nbsp;"` it'll render the nonbreaking space instead of the escape sequence. Hope this helps!

**Comment by jasonmahr** [comment]
Hey guys, I made a [package](https://github.com/jasonmahr/html-escape-sequences/tree/1.0.0) that solves this. If you use `text' "&nbsp;"` it'll render the non-breaking space instead of the escape sequence. Also works with `"&#160;"` and `"&#x000A0;"`. Hope this helps!
